hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1241,7 +1241,7 @@ public static int fromJavaDate(Date date) {
     return julianDays;
   }
 
-  private static Schema getActualSchemaFromUnion(Schema schema, Object data) {
+  public static Schema getActualSchemaFromUnion(Schema schema, Object data) {
```

Comment:
Avoid using public if this is only for testing.  Use the annotation that makes this visible to testing.